firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/impl/ScreenshotTaker.java

```diff
@@ -93,19 +99,58 @@ Task<Bitmap> captureScreenshot() {
           // We only take the screenshot here because this will be called on the main thread, so we
           // want to do as little work as possible
           try {
-            View view = activity.getWindow().getDecorView().getRootView();
-            Bitmap bitmap =
-                Bitmap.createBitmap(view.getWidth(), view.getHeight(), Bitmap.Config.RGB_565);
-            Canvas canvas = new Canvas(bitmap);
-            view.draw(canvas);
-            return bitmap;
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+              return captureScreenshotOreo(activity);
+            } else {
+              return captureScreenshotLegacy(activity);
+            }
           } catch (Exception | OutOfMemoryError e) {
             throw new FirebaseAppDistributionException(
                 "Failed to take screenshot", Status.UNKNOWN, e);
           }
         });
   }
 
+  private static Bitmap getBitmapForScreenshot(Activity activity) {
+    View view = activity.getWindow().getDecorView().getRootView();
+    return Bitmap.createBitmap(view.getWidth(), view.getHeight(), Bitmap.Config.RGB_565);
+  }
+
+  private static Task<Bitmap> captureScreenshotLegacy(Activity activity) {
+    Bitmap bitmap = getBitmapForScreenshot(activity);
+    Canvas canvas = new Canvas(bitmap);
+    activity.getWindow().getDecorView().getRootView().draw(canvas);
+    return Tasks.forResult(bitmap);
+  }
+
+  @TargetApi(Build.VERSION_CODES.O)
+  private static Task<Bitmap> captureScreenshotOreo(Activity activity) {
+    Bitmap bitmap = getBitmapForScreenshot(activity);
+    TaskCompletionSource<Bitmap> taskCompletionSource = new TaskCompletionSource<>();
+    LogWrapper.getInstance().e("LKELLOGG: NEW PATH HERE");
+    try {
+      // PixelCopy can handle Bitmaps with Bitmap.Config.HARDWARE and is the recommended way of
+      // capturing screens moving forward
```

Comment:
nit: Not sure if we need this (entire) comment. Maybe just abbreviate to: "PixelCopy can handle Bitmap.Config.HARDWARE".